### PR TITLE
[feature](multi-catalog) support select current_catalog();

### PR DESCRIPTION
**Pull request by nextdreamblue**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by nextdreamblue** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.33 seconds
 stream load tsv:          440 seconds loaded 74807831229 Bytes, about 162 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230328044513_clickbench_pr_121608.html

**Comment by nextdreamblue** [comment]
run COMPILE

**Comment by nextdreamblue** [comment]
run COMPILE

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentCatalog.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.doris.nereids.types.VarcharType;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ScalarFunction 'CurrentCatalog'.
+ */
+public class CurrentCatalog extends ScalarFunction
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentCatalog.java:34]: u need to do constant fold in fe, like the database() function do
```
org.apache.doris.nereids.rules.expression.rewrite.rules.FoldConstantRuleOnFE#visitDatabase
```

Comment (nextdreamblue) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/CurrentCatalog.java:34]: > u need to do constant fold in fe, like the database() function do
> 
> ```
> org.apache.doris.nereids.rules.expression.rewrite.rules.FoldConstantRuleOnFE#visitDatabase
> ```

ok, let me fix code

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run p0

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
@morrySnow @morningman 
pls review

**Review by morningman — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**